Electron/src/Core/components/ThreeDRover.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from "react"
+import CSS from "csstype"
+import path from "path"
+import { rovecomm } from "../RoveProtocol/Rovecomm"
+import STLViewer from "./STLViewer"
+import { windows } from "../Window"
+
+const container: CSS.Properties = {
+  display: "flex",
+  flexDirection: "column",
+  fontFamily: "arial",
+  borderTopWidth: "28px",
+  borderColor: "#990000",
+  borderBottomWidth: "2px",
+  borderStyle: "solid",
+  padding: "5px",
+  alignItems: "center",
+  height: "calc(100% - 47px)",
+}
+const label: CSS.Properties = {
+  marginTop: "-10px",
+  position: "relative",
+  top: "24px",
+  left: "3px",
+  fontFamily: "arial",
+  fontSize: "16px",
+  zIndex: 1,
+  color: "white",
+}
+
+const RoverFile = path.join(__dirname, "../assets/Rover.stl")
+
+interface IProps {
+  style?: CSS.Properties
+  zoom: number
+}
+
+interface IState {
+  IMUData: number[]
+  id: string
+  width: number
+  height: number
+}
+
+class ThreeDRover extends Component<IProps, IState> {
+  static defaultProps = {
+    zoom: 20,
+  }
+
+  constructor(props: any) {
+    super(props)
+    this.state = {
+      IMUData: [0, 0, 0],
+      id: new Date().toLocaleTimeString(),
```

Comment:
why do we need the date for the 3d rover?

Reply:
This is a slightly hacky solution to ensure that each 3D rover has a unique ID. In order to find the 3D rover in the page to set it's size properly, it must have a unique ID, and in anticipation of RID being able to have 1 or more 3D rovers in addition to the one in RED, it needed some scheme to have a unique name that wasn't tied to the screen name. I figured it was unlikely that multiple 3D rover's would be created within the same second (although this does create some issues if you have multiple displayed when you hot reload, as they all are created at the exact same time).

Reply:
Could you keep a running static index instead (e.g. 1,2,3,4)?